nupic/algorithms/anomaly.py

```diff
@@ -167,10 +178,35 @@ def __str__(self):
   def __setstate__(self, state):
     """deserialization"""
     self.__dict__.update(state)
-
     if not hasattr(self, '_mode'):
       self._mode = Anomaly.MODE_PURE
     if not hasattr(self, '_movingAverage'):
       self._movingAverage = None
     if not hasattr(self, '_binaryThreshold'):
       self._binaryThreshold = None
+
+
+  def getComputeFn(self, mode):
```

Comment:
The name of the function `getComputeFn` doesn't fully reflect what it does and that it is private; it doesn't just return compute function; there are side-effects in it, such as `self._likelihood = ...`. This function shouldn't exist at all; as discussed in one of my comments about the `compute` method, we shouldn't be calling it for every compute. The logic probably belongs in the constructor, where it also records the computation callable in a member variable for use by the `compute` method instead of calling `getComputeFn`.